presto-connector-tests/src/main/java/com/facebook/presto/connector/unittest/CreatesSchemas.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.unittest;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.SchemaTableName;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Closer;
+
+import java.io.Closeable;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+public interface CreatesSchemas
+        extends SPITest
+{
+    @Override
+    default SchemaTableName tableInDefaultSchema(String tableName)
```

Comment:
As I said somewhere else, I would put this somewhere else out of tests. Like `PlanBuilder` used in rules tests. Because:
 - it does not follow SRP
 - name space of test methods is mixed with utility methods
 - connector may override it (by design?)

Reply:
Relevant bit of the discussion @losipiuk and I had offline:

@lospiuk:
> Ok - so maybe we should have a `withSchema` method in `SPITest` which is called by `withSchemas`
> and have no default implementation of `withSchema`
> and make connector tests provide it?
> The for PSQL it may ignore the schema from table definition
> i.e. schema it gets as parameter and just be a noop

@ebd2:
> We can do that, but there are really only 2 sensible implementations. create and drop (Hive) or validate (postgresql).

@losipiuk:
> and for hive it would create schema at beginning and drop at the and after calling functor argument
> then maybe those two sensible implementations should be a superinterfaces of SPITest
> and PSQLConnectorTest would extend one
> and HiveConnectorTest the other
> the `SingleSchemaValidatingSPITest` would require extending classes
> to provide implementation of `getExpectedTableSchema`

For e.g. PostgreSQL, the implementations of `tableInDefaultSchema` and `withSchemas` are coupled and should be implemented together. See [ValidatesSchema](https://github.com/prestodb/presto/pull/8844/commits/1ce87f9772833f0bd3ea5bc670d14fbee4710410#diff-37e41362f567522fa5700026d3e3118f)

For Hive, or anything else that supports creating schemas, there's no reason to care what the schema name in the `SchemaTableName` returned by `tableInDefaultSchema` is, and so there's no reason to override it.

If you do care about the schema name, you're writing a different kind of test.